### PR TITLE
Fix Test a Pipeline not reading zip entries in certain cases

### PR DESCRIPTION
**Pull request by Matthbo**

### CODE REVIEW

File: console/backend/src/main/java/org/frankframework/management/web/TestPipeline.java [hunk 1]
```diff
@@ -150,30 +150,17 @@ private String processZipFile(InputStream file, RequestMessageBuilder builder) t
...
-				result.append(name);
-				result.append(": ");
-				result.append(BusMessageUtils.getHeader(response, ResponseMessageBase.STATE_KEY));
-				result.append("\n");
-			}
-			archive.closeEntry();
+		ZipEntry zipEntry;
+		while ((zipEntry = archive.getNextEntry()) != null) {
+			String name = zipEntry.getName();
+			String currentMessage = XmlEncodingUtils.readXml(StreamUtil.streamToBytes(StreamUtil.dontClose(archive)), null);
```
Comment (jkosternl) [inline console/backend/src/main/java/org/frankframework/management/web/TestPipeline.java:156]: Is this the easiest way to read from a stream into a String?
What about e.g. `String text = new String(inputStream.readAllBytes());`
or `String currentMessage = XmlEncodingUtils.readXml(archive, "UTF-8");`

**Review by jkosternl — CHANGES_REQUESTED** [review]

File: console/backend/src/main/java/org/frankframework/management/web/TransactionalStorage.java [hunk 2]
```diff
@@ -171,12 +172,13 @@ public void write(OutputStream out) throws IOException {
 						}
 
 						String payload = (String) message.getPayload();
+						byte[] payloadBytes = StringUtils.getBytes(payload, StandardCharsets.UTF_8);
```
[the remark this answers is not in the document]
Comment (Matthbo) [reply console/backend/src/main/java/org/frankframework/management/web/TransactionalStorage.java:175]: Is there a big difference to which one should be used?
I know this one exists but I've been corrected before by Niels when not using our own utils function for something

[the remark this answers is not in the document]
Comment (nielsm5) [reply console/backend/src/main/java/org/frankframework/management/web/TransactionalStorage.java:175]: Yea but this (StringUtils) is not from us 😄 
That's about binary -> character data conversions so we can use our internals to see if there's a bom persent.

File: console/backend/src/main/java/org/frankframework/management/web/TestPipeline.java (hunk 1 above)
Comment (nielsm5) [reply console/backend/src/main/java/org/frankframework/management/web/TestPipeline.java:156]: You must not close the stream!

File: console/backend/src/main/java/org/frankframework/management/web/TestPipeline.java [hunk 3]
```diff
@@ -150,30 +150,17 @@ private String processZipFile(InputStream file, RequestMessageBuilder builder) t
 		StringBuilder result = new StringBuilder();
 
 		ZipInputStream archive = new ZipInputStream(file);
```
Comment (nielsm5) [inline console/backend/src/main/java/org/frankframework/management/web/TestPipeline.java:152]: Maybe you can use a try-with-resources here.

**Review by nielsm5 — APPROVED** [review]

File: console/backend/src/main/java/org/frankframework/management/web/TestPipeline.java [hunk 4]
```diff
@@ -149,22 +149,11 @@ private Response testPipelineResponse(String payload, String state, String messa
...
-						break;
-					}
-					rb+=chunk;
-				}
-				String currentMessage = XmlEncodingUtils.readXml(b, null);
+		try(ZipInputStream archive = new ZipInputStream(file)) {
+			ZipEntry zipEntry;
+			while ((zipEntry = archive.getNextEntry()) != null) {
+				String name = zipEntry.getName();
+				String currentMessage = XmlEncodingUtils.readXml(StreamUtil.streamToBytes(StreamUtil.dontClose(archive)), null);
```
[the remark this answers is not in the document]
Comment (jkosternl) [reply console/backend/src/main/java/org/frankframework/management/web/TestPipeline.java:156]: Niels says the unit test already validates it, so it should be correct.

**Review by jkosternl — APPROVED** [review]

**Review by nielsm5 — APPROVED** [review]

**Pull request merged**